screens/AuthStack.tsx

```diff
@@ -0,0 +1,150 @@
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import { StatusBar } from "expo-status-bar";
+import {
+  View,
+  Text,
+  Button,
+  TextInput,
+  TouchableOpacity,
+  Alert,
+} from "react-native";
+import { useForm, FieldError } from "react-hook-form";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useCallback, useEffect, useState } from "react";
+import { styles } from "./styles";
+import { supabase } from "../supabase-service";
+type FormInputs = {
+  email: string;
+  password: string;
+};
+
+export const loginSchema = yup.object().shape({
+  email: yup
+    .string()
+    .email("Invalid Email Format")
+    .required("Email is a required field"),
+  password: yup.string().required("Password is a required field"),
+});
+
+export const ErrorText = ({
+  name,
+  errors,
+}: {
+  name: string;
+  errors: Record<string, any>;
+}) => {
+  return (
+    <View>
+      {errors[name] && (
+        <Text style={{ color: "red" }}>{errors?.[name]?.message}</Text>
+      )}
+    </View>
+  );
+};
+
+const ErrorAlert = ({ title, message }: { title: string; message: string }) =>
+  Alert.alert(title, message, [
+    { text: "OK", onPress: () => console.log("Ok pressed") },
```

Comment:
Hopefully we can avoid having `console.log`s in our code!
```suggestion
    { text: "OK" },
```